cni/network/network.go

```diff
@@ -976,21 +976,14 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
 
 	// Query the network.
 	if nwInfo, err = plugin.nm.GetNetworkInfo(networkID); err != nil {
-
 		if !nwCfg.MultiTenancy {
```

Comment:
instead of ignoring all the non-nil errors, we should only ignore if the error is of type NetworkNotFoundError
https://github.com/microsoft/hcsshim/blob/7fa8bda4e6ba503caf0d53d0a4ee99b9a64ceed8/internal/hns/hns.go#L21

If the error is not of type NetworkNotFound, we should return error.

we do have similar code for hns v2 path (hcn). I believe this should work for v1 as well.
https://github.com/Azure/azure-container-networking/blob/96bec09d413e53e06e3776bbab3a2d6bd0e6ad31/cns/hnsclient/hnsclient_windows.go#L287

Reply:
GetNetworkInfo checks cni state and return network. It doesn't call hns. If there is state it returns that else it returns error. The only error is not found